packages/HilltopCrawler/src/main/kotlin/nz/govt/eop/hilltop_crawler/fetcher/FetchTaskProcessor.kt

```diff
@@ -0,0 +1,158 @@
+
+      if (xmlContent == null) {
+        handleTaskErrorRequeue(
+            taskToProcess, fetchedAt, FETCH_ERROR, Duration.ofMinutes(5), Duration.ofHours(1))
+        return
+      }
+
+      val isErrorXml =
+          try {
+            parsers.isHilltopErrorXml(xmlContent)
+          } catch (e: JsonParseException) {
+            logger.warn(e) { "Failed to parse content [$xmlContent]" }
+            true
+          }
+      if (isErrorXml) {
+        handleTaskErrorRequeue(
+            taskToProcess, fetchedAt, HILLTOP_ERROR, Duration.ofDays(1), Duration.ofDays(30))
+        return
+      }
+
+      val taskMapper =
+          try {
+            when (taskToProcess.requestType) {
+              SITES_LIST ->
+                  SitesListTaskMapper(
+                      source,
+                      taskToProcess.fetchUri,
+                      fetchedAt,
+                      xmlContent,
+                      parsers.parseSitesResponse(xmlContent))
+              MEASUREMENTS_LIST ->
+                  MeasurementsListTaskMapper(
+                      source,
+                      taskToProcess.fetchUri,
+                      fetchedAt,
+                      xmlContent,
+                      parsers.parseMeasurementsResponse(xmlContent))
+              MEASUREMENT_DATA ->
+                  MeasurementDataTaskMapper(
+                      source,
+                      taskToProcess.fetchUri,
+                      fetchedAt,
+                      xmlContent,
+                      parsers.parseMeasurementValuesResponse(xmlContent))
+            }
+          } catch (e: JsonMappingException) {
+            logger.warn(e) { "Failed to parse content [${xmlContent}]" }
+            handleTaskErrorRequeue(
+                taskToProcess, fetchedAt, PARSE_ERROR, Duration.ofDays(1), Duration.ofDays(30))
+            return
+          }
+
+      if (taskMapper.contentHash != taskToProcess.previousDataHash) {
+        taskMapper.buildNewTasksList().forEach(db::createFetchTask)
+        taskMapper.buildKafkaMessage()?.let(kafkaClient::send)
+      }
+
+      handleTaskRequeue(
+          taskToProcess,
+          fetchedAt,
+          taskMapper.contentHash,
+          if (taskMapper.contentHash == taskToProcess.previousDataHash) UNCHANGED else SUCCESS,
+          taskMapper.determineNextFetchAt())
+    } catch (e: Exception) {
+      // This is a catch-all for any errors that occur while processing a task.
+      logger.error(e) {
+        "Failed to process task [${taskToProcess.id}] with url [${taskToProcess.fetchUri}]"
+      }
+      // We want to make sure if we fail to process a task, we reschedule it to run again
+      // with a delay so that we don't end up in a situation where we are constantly trying to
+      // process a task that is failing.
+      handleTaskErrorRequeue(
+          taskToProcess, fetchedAt, UNKNOWN_ERROR, Duration.ofDays(1), Duration.ofDays(7))
+    }
```

Comment:
Yeah, this is kinda relying on us responding to the error message that gets logged, I was on the fence about how long in case its caused by something non-transient. So didn't want it looping over the failure too quick.